src/client/producer.rs

```diff
@@ -307,12 +307,18 @@ where
     status_deagg: <A as aggregator::Aggregator>::StatusDeaggregator,
 }
 
+#[derive(Debug, Clone)]
+enum AggregatedError {
+    Aggregator(Arc<aggregator::Error>),
```

Comment:
I think this is an Arc of a box which seems a tad redundant?

Reply:
aggregated error is gone